libraries/ESP8266WiFi/src/include/ClientContext.h

```diff
@@ -325,6 +328,26 @@ class ClientContext
         return _write_from_source(new BufferedStreamDataSource<ProgmemStream>(stream, size));
     }
 
+    void keepAlive (uint16_t idle_sec = TCP_DEFAULT_KEEPALIVE_IDLE_SEC, uint16_t intv_sec = TCP_DEFAULT_KEEPALIVE_INTERVAL_SEC, uint8_t count = TCP_DEFAULT_KEEPALIVE_COUNT)
+    {
+        if (idle_sec && intv_sec && count) {
+            _pcb->so_options |= SOF_KEEPALIVE;
+            _pcb->keep_idle = (uint32_t)1000 * idle_sec;
+            _pcb->keep_intvl = (uint32_t)1000 * intv_sec;
+            _pcb->keep_cnt = count;
+        }
+        else
+            _pcb->so_options &= ~SOF_KEEPALIVE;
+    }
+
+    void getKeepAlive (uint16_t* idle_sec, uint16_t* intv_sec, uint8_t* count)
+    {
+        bool enabled = _pcb->so_options & SOF_KEEPALIVE;
+        if (idle_sec) *idle_sec = enabled? (_pcb->keep_idle + 500) / 1000: 0;
```

Comment:
Indentation (body of the if statement on a new line)

Reply:
will fix